FirmataMarshaller.cpp

```diff
@@ -173,17 +216,20 @@ const
  * when using the ANALOG_MESSAGE. The maximum value of the ANALOG_MESSAGE is limited to 14 bits
  * (16384). To increase the pin range or value, see the documentation for the EXTENDED_ANALOG
  * message.
- * @param pin The analog pin to send the value of (limited to pins 0 - 15).
+ * @param pin The analog pin to which the value is sent.
  * @param value The value of the analog pin (0 - 1024 for 10-bit analog, 0 - 4096 for 12-bit, etc).
- * The maximum value is 14-bits (16384).
  */
 void FirmataMarshaller::sendAnalog(uint8_t pin, uint16_t value)
 const
 {
   if ( (Stream *)NULL == FirmataStream ) { return; }
-  // pin can only be 0-15, so chop higher bits
-  FirmataStream->write(ANALOG_MESSAGE | (pin & 0xF));
-  sendValueAsTwo7bitBytes(value);
+
+  if ( (0xF >= pin) && (0x3FFF >= value) ) {
```

Comment:
Ha! Yeah, I used to be teaching assistant, so I always taught the kids to put it on the left because you can't accidentally set it when trying to compare.